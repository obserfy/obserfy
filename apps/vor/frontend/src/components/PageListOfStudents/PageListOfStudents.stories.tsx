import { FC } from "react"
import PageListOfStudents from "./PageListOfStudents"

export default {
  title: "Core/PageListOfStudents",
  component: PageListOfStudents,
  parameters: {
    componentSubtitle: "Just a simple PageListOfStudents",
  },
}

export const Basic: FC = () => <PageListOfStudents />
