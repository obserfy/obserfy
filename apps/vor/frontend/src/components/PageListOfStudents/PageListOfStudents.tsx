import { FC, Fragment, useState } from "react"
import { Box, Card, Flex, Image } from "theme-ui"
import { t, Trans } from "@lingui/macro"
import BackNavigation from "../BackNavigation/BackNavigation"
import { ADMIN_URL } from "../../routes"
import Typography from "../Typography/Typography"
import { useGetAllStudents } from "../../hooks/api/students/useGetAllStudents"
import SearchBar from "../SearchBar/SearchBar"
import Pill from "../Pill/Pill"
import AlertDialog from "../AlertDialog/AlertDialog"
import { usePatchStudentApi } from "../../hooks/api/students/usePatchStudentApi"
import StudentPicturePlaceholder from "../StudentPicturePlaceholder/StudentPicturePlaceholder"

export const PageListOfStudents: FC = () => {
  const students = useGetAllStudents()
  const [searchTerm, setSearchTerm] = useState("")
  const filteredStudents = students.data?.filter((student) =>
    student.name.match(new RegExp(searchTerm, "i"))
  )
  return (
    <Box
      sx={{
        flexDirection: "column",
        maxWidth: "maxWidth.md",
      }}
      mx="auto"
    >
      <BackNavigation to={ADMIN_URL} text={t`Settings`} />
      <Typography.H5 m={3} sx={{ lineHeight: 1 }}>
        <Trans>All Students</Trans>
      </Typography.H5>
      <Box px={3} pb={2} pt={2}>
        <SearchBar
          sx={{ width: "100%" }}
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
      </Box>
      {filteredStudents?.map(({ id, name, active, profileImageUrl }) => (
        <StudentCard
          key={id}
          profilePicUrl={profileImageUrl}
          name={name}
          active={active}
          studentId={id}
        />
      ))}
    </Box>
  )
}

const StudentCard: FC<{
  studentId: string
  active: boolean
  name: string
  profilePicUrl?: string
}> = ({ studentId, active, name, profilePicUrl }) => {
  const { mutateAsync } = usePatchStudentApi(studentId)
  const [showStatusDialog, setShowStatusDialog] = useState(false)
  const saveStatus = async () => {
    try {
      await mutateAsync({ active: !active })
      setShowStatusDialog(false)
    } catch (e) {
      Sentry.captureException(e)
    }
  }
  const setActiveText = t`Set as active?`
  const setInactiveText = t`Set as inactive?`
  return (
    <Fragment>
      <Card
        p={3}
        mx={[0, 3]}
        mb={[0, 2]}
        sx={{
          backgroundColor: ["background", "surface"],
          borderRadius: [0, "default"],
          cursor: "pointer",
          boxShadow: ["none", "low"],
          display: "flex",
          alignItems: "center",
        }}
        onClick={() => setShowStatusDialog(true)}
      >
        {profilePicUrl ? (
          <Image
            src={profilePicUrl}
            sx={{
              width: 32,
              height: 32,
              borderRadius: "circle",
            }}
          />
        ) : (
          <StudentPicturePlaceholder />
        )}
        <Flex sx={{ flexDirection: "column", alignItems: "start" }}>
          <Typography.Body ml={3} mb={2} sx={{ lineHeight: 1.6 }}>
            {name}
          </Typography.Body>
          <Pill
            style={{ width: "fit-content", cursor: "pointer" }}
            text={active ? t`active` : t`inactive`}
            backgroundColor={active ? "primary" : "warning"}
            color={active ? "onPrimary" : "onWarning"}
            mx={3}
          />
        </Flex>
      </Card>
      {showStatusDialog && (
        <AlertDialog
          title={active ? setInactiveText : setActiveText}
          negativeText={t`Cancel`}
          positiveText={t`Yes`}
          body={`Are you sure you want to set ${name} as ${
            active ? "inactive" : "active"
          }?`}
          onNegativeClick={() => setShowStatusDialog(false)}
          onPositiveClick={() => saveStatus()}
        />
      )}
    </Fragment>
  )
}

export default PageListOfStudents
