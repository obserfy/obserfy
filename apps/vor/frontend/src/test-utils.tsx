/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
