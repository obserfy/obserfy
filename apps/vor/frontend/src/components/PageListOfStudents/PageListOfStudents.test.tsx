import { render } from "../../test-utils"
import { Basic } from "./PageListOfStudents.stories"

describe("PageListOfStudents", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
