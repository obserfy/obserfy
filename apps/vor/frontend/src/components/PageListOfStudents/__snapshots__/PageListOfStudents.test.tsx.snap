// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageListOfStudents should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 700px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin: 16px;
  line-height: 1;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 13px;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: -33px;
  margin-left: 8px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  width: 24px;
  height: 24px;
  z-index: 2;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: var(--theme-ui-colors-surface);
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  padding-left: 40px;
  font-size: 14px;
  width: 100%;
  width: 100%;
  width: 100%;
}

.emotion-10:autofill,
.emotion-10:autofill:hover,
.emotion-10:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-10:autofill:first-line,
.emotion-10:autofill:hover:first-line,
.emotion-10:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-10:-webkit-autofill,
.emotion-10:-webkit-autofill:hover,
.emotion-10:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-10:-webkit-autofill:first-line,
.emotion-10:-webkit-autofill:hover:first-line,
.emotion-10:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-10:hover,
.emotion-10:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-10::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-10::-moz-placeholder {
  font-style: italic;
}

.emotion-10:-ms-input-placeholder {
  font-style: italic;
}

.emotion-10::placeholder {
  font-style: italic;
}

.emotion-10:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/admin"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            Settings
          </p>
        </div>
      </a>
      <h5
        class="emotion-5"
      >
        All Students
      </h5>
      <div
        class="emotion-6"
      >
        <label
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            <div
              class="emotion-9"
            />
            <input
              class="emotion-10"
              name="name"
              placeholder="Search"
              value=""
            />
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
